plugins/xvendor-xbasename/README.md

```diff
@@ -0,0 +1,34 @@
+---
+title: Plugin: Xvendor XBasename
+---
+
+## Introduction
+
+The XBasename is a bla bla bla.
+
+## Firmware Format
+
+The daemon will decompress the cabinet archive and extract a firmware blob in
+a packed binary file format.
+
+This plugin supports the following protocol ID:
+
+* `com.xvendor.xbasename`
+
+## GUID Generation
+
+These devices use the standard USB DeviceInstanceId values, e.g.
+
+* `XSUBSYS\ID_XXX`
+
+## Update Behavior
+
+The device is updated by bla bla bla.
+
+## Vendor ID Security
+
+The vendor ID is set from the USB vendor, in this instance set to `USB:0x273F`
+
+## External Interface Access
+
+This plugin requires read/write access to `/dev/bus/usb`.
```

Comment:
This stuff I don't think we should auto-populate.  It really depends on the type of plugin, and it's just one more thing for people to not know how to write.  Or we should have a few example types and pick the one that makes sense from an argparse argument.

Reply:
I think I should move the entire thing to something like the jinja templating engine -- then we can do %if and stuff like that.

Reply:
Something like that? if that looks better (and I think it does) then I'll add a few more `%if`'s and make it work better for HID and Udev types.

Reply:
Oh yeah;  Much better.  With templating like this though, I think we need to have a CI job that is going to call it with some arguments and make sure that it spits out real code that can compile and that we're not just giving people a headache to start out with.

Reply:
What about this? i.e. checking in the .c/h files, having a default off meson option that we turn regenerate and turn on in one CI target?

Reply:
I did that.